src/main/java/uk/gov/companieshouse/overseasentitiesapi/Interceptor/LoggingInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.overseasentitiesapi.Interceptor;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.ERIC_REQUEST_ID_KEY;
+
+@Component
+public class LoggingInterceptor implements HandlerInterceptor {
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        Long startTime = System.currentTimeMillis();
+        request.getSession().setAttribute("start-time", startTime);
```

Comment:
Could define a class private constant with the "start-time" string as it appears here and also in method below.